internal/uploader.go

```diff
@@ -22,6 +22,7 @@ type Uploader struct {
 	deltaFileManager     *DeltaFileManager
 	archiveStatusManager DataFolder
 	Failed               atomic.Value
+	tarSize              *int64
```

Comment:
is there any reason why we use a pointer, is it possible to simply pass by value?

Reply:
It's not possible because of races.